cmd/kubeadm/app/cmd/phases/init/uploadcerts.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"github.com/pkg/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	nodebootstraptokenphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/uploadcerts"
+)
+
+type uploadCertsData interface {
+	Client() (clientset.Interface, error)
+	UploadCerts() bool
+	CertificateKey() string
+	CertificateDir() string
+}
+
+// NewUploadCertsPhase returns the uploadCerts phase
+func NewUploadCertsPhase() workflow.Phase {
+	return workflow.Phase{
+		Name:  "upload-certs",
+		Short: "Upload certificates to kubeadm-certs",
+		Long:  cmdutil.MacroCommandLongDescription,
+		Run:   runUploadCerts,
+		InheritFlags: []string{
+			options.CfgPath,
+			options.UploadCerts,
+			options.CertificateKey,
+		},
+	}
+}
+
+func runUploadCerts(c workflow.RunData) error {
+	data, ok := c.(uploadCertsData)
+	if !ok {
+		return errors.New("upload-certs phase invoked with an invalid data struct")
+	}
+
+	//skip upload certs.
+	if !data.UploadCerts() {
+		return nil
+	}
+	client, err := data.Client()
+	if err != nil {
+		return err
+	}
+
+	// Create the short lived bootstrap token
+	tokens, err := generateShortLivedToken()
```

Comment:
I think that we should move the details about the TTL token away from `cmd/phases/init/uploadcerts.go`, and move to it `phases/uploadcerts/uploadcerts.go`